File: old_code/src/utils/IProductionPart.ts
/**
 * Enum representing the type of production part.
 */
export enum ProductionPartType {
  Terminal,
  NonTerminal
}

/**
 * Interface for production parts (terminals and non-terminals).
 */
export interface IProductionPart {
  /**
   * Gets the name of the production part.
   */
  getName(): string;

  /**
   * Gets the type of the production part.
   */
  getType(): ProductionPartType;

  /**
   * Creates a string representation of this production part.
   */
  toString(): string;
}

File: src/StepParser/Core/IProductionPart.cs
namespace StepParser.Core;

/// <summary>
/// Represents the type of a production part.
/// </summary>
public enum ProductionPartType
{
    Terminal,
    NonTerminal,
    Optional,
    ZeroOrMore,
    OneOrMore,
    Group
}

/// <summary>
/// Interface for production parts in a grammar rule.
/// </summary>
public interface IProductionPart
{
    /// <summary>
    /// Gets the type of this production part.
    /// </summary>
    ProductionPartType Type { get; }

    /// <summary>
    /// Gets the name of this production part.
    /// </summary>
    string Name { get; }

    /// <summary>
    /// Gets a value indicating whether order is important for this part.
    /// </summary>
    bool IsOrderImportant { get; }
}